### PR TITLE
Add significance documentation

### PR DESCRIPTION
**Pull request by MariusArhaug**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by kkraune** [comment]
Great! not sure why we have two files in reference for this instead of just one

File: en/operations-selfhosted/vespa-cmdline-tools.html [hunk 1]
```diff
@@ -1910,6 +1910,56 @@ <h2 id="vespa-set-node-state">vespa-set-node-state</h2>
 
 <!--h2 id="vespa-slobrok-cmd">vespa-slobrok-cmd</h2-->
 
+<h2 id="vepsa-signficance">vespa-significance</h2>
+<p><code>vepsa-signficance</code> is a tool that generates a significance model file based on <a href="">this</a> file format. Its input is a <code>vespa-feed</code> file.
```
Comment (bjorncs) [inline en/operations-selfhosted/vespa-cmdline-tools.html:1914]: Update placeholder link

Comment (bjorncs) [inline en/operations-selfhosted/vespa-cmdline-tools.html:1914]: `s/vepsa-significance/vespa-significance` ;)

Comment (bjorncs) [reply en/operations-selfhosted/vespa-cmdline-tools.html:1914]: Link to https://docs.vespa.ai/en/reference/document-json-format.html and specify that the input file is in the "Document JSON format"

File: en/operations-selfhosted/vespa-cmdline-tools.html [hunk 2]
```diff
@@ -1910,6 +1910,56 @@ <h2 id="vespa-set-node-state">vespa-set-node-state</h2>
...
+    <tr>
+      <th>-h, --help</th>
+      <td>
+        Help text
+      </td>
+    </tr>
+    <tr>
+      <th>-i, --input &lt;input file&gt;</th>
+      <td>
+        Vespa dump file to be used for generating the significance model
```
Comment (bjorncs) [inline en/operations-selfhosted/vespa-cmdline-tools.html:1938]:
```suggestion
        Vespa feed file to be used for generating the significance model
```

File: en/operations-selfhosted/vespa-cmdline-tools.html [hunk 3]
```diff
@@ -1910,6 +1910,56 @@ <h2 id="vespa-set-node-state">vespa-set-node-state</h2>
...
+      </td>
+    </tr><tr>
+      <th>-o, --out &lt;output file&gt;</th>
+      <td>
+        Output file for the significance model
+      </td>
+    </tr><tr>
+      <th> -f, --field &lt;field&gt;</th>
+      <td>
+        Name of the text field to be used for tokenization
```
Comment (bjorncs) [inline en/operations-selfhosted/vespa-cmdline-tools.html:1948]:
```suggestion
        Name of the text field to be used for significance model 
```

File: en/operations-selfhosted/vespa-cmdline-tools.html [hunk 4]
```diff
@@ -1910,6 +1910,56 @@ <h2 id="vespa-set-node-state">vespa-set-node-state</h2>
...
+      </td>
+    </tr><tr>
+      <th> -l, --language &lt;language&gt;</th>
+      <td>
+        Language of the text field, must be a valid language code from the <a href="https://www.rfc-editor.org/rfc/rfc5646">RFC5646</a> standard.
+      </td>
+    </tr><tr>
+      <th> -d, --doc-type &lt;doc-id&gt;</th>
+      <td>
+          Document type identifier for the dump file
```
Comment (bjorncs) [inline en/operations-selfhosted/vespa-cmdline-tools.html:1958]: Please include more details on the purpose of this parameter.

File: en/reference/schema-reference.html [hunk 5]
```diff
@@ -2481,6 +2486,15 @@ <h2 id="onnx-model">onnx-model</h2>
 </table>
 <p>For more details including examples, see <a href="../onnx.html">ranking with ONNX models.</a></p>
 
+<h2 id="significance">significance</h2>
+<p>
+Constrained in <a href="#rank-profile">rank-profile</a>. True or false. By default this is false. When enabled Vespa will use the significance calculation based on provided significance models in the service.xml for the rank-profile it is defined in.
```
Comment (bjorncs) [inline en/reference/schema-reference.html:2491]: s/Constrained/Contained

File: en/reference/services-search.html [hunk 6]
```diff
@@ -328,6 +329,20 @@ <h2 id="renderer">renderer</h2>
           bundle="the name in &lt;artifactId&gt; in pom.xml" /&gt;
 </pre>
 
+<h2 id="significance">significance</h2>
+<p>
+The significance tag can include multiple models. Their order determines the model precedence for a given language, with the last element having the highest. The models' document frequency is used to set a token's significance value based on the inverse document frequency (IDF). To enable the use of these models, the schema needs to have a rank-profile field with the <em>significance</em> element and the  <em>use-model</em> flag set to <em>true</em>.
```
Comment (bjorncs) [inline en/reference/services-search.html:334]:
```suggestion
The significance element can include multiple models. Their order determines the model precedence for a given language, with the last element having the highest. The models' document frequency is used to set a token's significance value based on the inverse document frequency (IDF). To enable the use of these models, the schema needs to have a rank profile with the <em>significance</em> element and the <em>use-model</em> set to <em>true</em>.
```

File: en/reference/services-search.html [hunk 7]
```diff
@@ -328,6 +329,20 @@ <h2 id="renderer">renderer</h2>
...
+</p>
+
+<p>Example of significance model with multiple models. These models are either provided by <em>Vespa</em> or can be generated with the <a href="vespa-cmdline-tools.html#vespa-significance">vepsa-signficance</a> cli. </p>
```
Comment (bjorncs) [inline en/reference/services-search.html:337]: Add a link to https://docs.vespa.ai/en/reference/config-files.html#model which documents the semantics of the `model-id`,`url` and `path` attributes. The first sentence can be simplified to e.g. "Example with multiple model files". There is another "vepsa" typo here ;)

**Review by bjorncs — COMMENTED** [review]
I'll review the two new documents later (`significance.html`/`significance-reference.html`). Here's some feedback on the other files.

File: en/reference/schema-reference.html [hunk 8]
```diff
@@ -2481,6 +2486,15 @@ <h2 id="onnx-model">onnx-model</h2>
 </table>
 <p>For more details including examples, see <a href="../onnx.html">ranking with ONNX models.</a></p>
 
+<h2 id="significance">significance</h2>
+<p>
+Contained in <a href="#rank-profile">rank-profile</a>. True or false. By default this is false. When enabled Vespa will use the significance calculation based on provided significance models in the service.xml for the rank-profile it is defined in. 
+<pre>
+significance {
```
Comment (geirst) [inline en/reference/schema-reference.html:2493]: This section will be extended with more configuration later (e.g. which formula to use to calculate the significance). Please restructure similar to https://docs.vespa.ai/en/reference/schema-reference.html#onnx-model.

File: en/reference/services-search.html [hunk 9]
```diff
@@ -328,6 +329,44 @@ <h2 id="renderer">renderer</h2>
           bundle="the name in &lt;artifactId&gt; in pom.xml" /&gt;
 </pre>
 
+<h2 id="significance">significance</h2>
+<p>
+The significance element can include multiple models. Their order determines the model precedence for a given language, with the last element having the highest. The models' document frequency is used to set a token's significance. To enable the use of these models, the schema needs to have a rank-profile with the <em>significance</em> element and the  <em>use-model</em> set to <em>true</em>.
```
Comment (geirst) [inline en/reference/services-search.html:334]: This section should link to both `significance.html` and `reference/schema-reference.html#significance`.

File: en/reference/services-search.html [hunk 10]
```diff
@@ -328,7 +329,63 @@ <h2 id="renderer">renderer</h2>
...
+
+<h2 id="model">model</h2>
+<p>
+Contained in <a href="#siginficance">significance</a>.
+Specifies <a href="../significance.html#global-significance-model">global significance model</a>.
+Models are identified by <code>model-id</code> or by providing <code>url</code> or <code>path</code> to a model file in the application package.
+</p>
+<p>
+Models with <code>model-id</code> are provided by <em>Vespa</em>.
+So far the only model available is <em>wikimedia</em>, which is generated from English Wikipedia.
```
Comment (geirst) [inline en/reference/services-search.html:367]: I am not sure what will be the `id` of this model. I suggest we remove this section until those details are decided.

**Review by geirst — COMMENTED** [review]
Overall looks good, some comments to look at.

File: en/reference/services-search.html (hunk 10 above)
Comment (glebashnik) [reply en/reference/services-search.html:367]: Do you want to remove mentioning of model-id attribute or just the wikimedia example?

Comment (geirst) [reply en/reference/services-search.html:367]: We can remove the wikimedia example for now.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**